pkg/api/accesscontrol.go

```diff
@@ -37,10 +30,9 @@ var (
 	ScopeProvisionersDatasources   = ac.Scope("provisioners", "datasources")
 	ScopeProvisionersNotifications = ac.Scope("provisioners", "notifications")
 
-	ScopeDatasourcesAll = ac.Scope("datasources", "*")
-	ScopeDatasourceID   = ac.Scope("datasources", "id", ac.Parameter(":id"))
-	ScopeDatasourceUID  = ac.Scope("datasources", "uid", ac.Parameter(":uid"))
-	ScopeDatasourceName = ac.Scope("datasources", "name", ac.Parameter(":name"))
+	ScopeDatasourceID   = ac.GetResourceScope(ac.ScopeDatasourcesRoot, ac.Parameter(":id"))
+	ScopeDatasourceUID  = ac.GetResourceScopeUID(ac.ScopeDatasourcesRoot, ac.Parameter(":uid"))
+	ScopeDatasourceName = ac.GetResourceScopeName(ac.ScopeDatasourcesRoot, ac.Parameter(":name"))
```

Comment:
nit: let's also move these data source scopes to `pkg/services/accesscontrol/models.go` to have all data source related actions and scopes in one place.

Reply:
The reason I left them close to where they're used is the `Parameter`, which can be different in different use cases. For example, in alerting API it is `:Recipient` not `:id`.